c3/generator/devices.py

```diff
@@ -193,17 +243,20 @@ def __init__(self, **props):
         self.ts = None
         self.sampling_method = props.pop("sampling_method", "nearest")
 
-    def process(self, instr, chan, awg_signal):
+    def process(self, instr: Instruction, chan: str, awg_signal: dict) -> dict:
         """Resample the awg values to higher resolution.
 
         Parameters
         ----------
-        awg_signal: tf.Tensor
-            Bandwith-limited, low-resolution AWG signal.
-        t_start: np.float64
-            Beginning of the signal.
-        t_end: np.float64
-            End of the signal.
+        Parameters
```

Comment:
Similar examples in here would also be useful

Reply:
This might need to be reworked or at least talked about before we finalize it.